hypershift/hosted-wrapper.py

```diff
@@ -259,6 +267,97 @@ def _cluster_load(kubeconfig, my_path, hosted_cluster_name, load_duration, jobs,
     load_process_stdout, load_process_stderr = load_process.communicate()
 
 
+def _get_must_gather(my_path, cluster_name):
+    cluster_path = my_path + "/" + cluster_name
+    myenv = os.environ.copy()
+    myenv["KUBECONFIG"] = cluster_path + "/kubeconfig"
+    logging.info('Gathering facts of hosted cluster %s' % cluster_name)
+    must_gather_command = ["oc", "adm", "must-gather", "--dest-dir", cluster_path + "/must_gather"]
+    logging.debug(must_gather_command)
+    must_gather_log = open(cluster_path + '/must_gather.log', 'w')
+    must_gather_process = subprocess.Popen(must_gather_command, stdout=must_gather_log, stderr=must_gather_log, env=myenv)
+    must_gather_stdout, must_gather_stderr = must_gather_process.communicate()
+    if must_gather_process.returncode != 0:
+        logging.error("Failed to obtain must-gather from %s" % cluster_name)
+        return 1
+    logging.info('Compressing must gather artifacts on %s file' % cluster_path + "/must_gather.tar.gz")
+    must_gather_compress_command = ["tar", "czvf", "must_gather.tar.gz", cluster_path + "/must_gather"]
+    logging.debug(must_gather_compress_command)
+    must_gather_compress_process = subprocess.Popen(must_gather_compress_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=myenv)
+    must_gather_compress_stdout, must_gather_compress_stderr = must_gather_compress_process.communicate()
+    if must_gather_compress_process.returncode != 0:
+        logging.error("Failed to compress must-gather of %s cluster from %s to %s" % (cluster_name, cluster_path + "/must_gather", cluster_path + "must_gather.tar.gz"))
+        return 1
+    logging.info('Deleting non-compressed must-gather files of hosted cluster %s' % cluster_name)
+    must_gather_delete_command = ["rm", "-rf", cluster_path + "/must_gather"]
+    logging.debug(must_gather_delete_command)
+    must_gather_delete_process = subprocess.Popen(must_gather_delete_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=myenv)
+    must_gather_delete_stdout, must_gather_delete_stderr = must_gather_delete_process.communicate()
+    if must_gather_delete_process.returncode != 0:
+        logging.error("Failed to delete non-compressed must-gather files of hosted cluster %s" % cluster_name)
+        return 1
```

Comment:
I don't think we should "fail" the test b/c we couldnt get a must gather as the actual workload may have run fine. Can we change these to logging.warn and just alert that it didnt work in the logs? Same for all the other ones that do a must gather

Reply:
It wont fail the test, that return is not used to determine the status. I need to return something, to stop the execution of the function, so it wont compress a file that not exists or something like that

I can change to return 0 and logging.warning, but that wont affect to the final result of the script

Reply:
ack. disregard then